packages/data-provider/src/utils/submissionUtils.ts

```diff
@@ -140,55 +142,70 @@ const utils = (dependencies: Dependencies) => {
 		},
 
 		/**
-		 * Run Schema Validation process
-		 * @param {SchemasDictionary} dictionary The dictionary to validate data with
-		 * @param {string} entityName The entity Name
-		 * @param {ReadonlyArray<DataRecord>} records An Array of the records to validate
-		 * @returns The result of the Schema validation
+		 * Checks if object is a Submission or a SubmittedData
+		 * @param {SubmittedDataReference | SubmissionReference} toBeDetermined
+		 * @returns {boolean}
 		 */
-		processSchemaValidation: async (
-			dictionary: SchemasDictionary,
-			entityName: string,
-			records: ReadonlyArray<DataRecord>,
-		): Promise<{ processedRecords: TypedDataRecord[]; schemaErrors: SchemaValidationError[] }> => {
-			const validRecords: any[] = [];
-			const schemaErrors: any[] = [];
+		determineIfIsSubmission: (
+			toBeDetermined: SubmittedDataReference | SubmissionReference,
+		): toBeDetermined is SubmissionReference => {
+			if ((toBeDetermined as SubmissionReference).type === MERGE_REFERENCE_TYPE.SUBMISSION) {
+				return true;
+			}
+			return false;
```

Comment:
Not a fan of the assertion, but if you must, at least the return can be simplified like so
```suggestion
			return (toBeDetermined as SubmissionReference).type === MERGE_REFERENCE_TYPE.SUBMISSION;
```